src/states/elder/mod.rs

```diff
@@ -492,7 +496,11 @@ impl Elder {
             return Err(RoutingError::UnknownConnection(pub_id));
         }
 
-        if let Some(signed_msg) = self.sig_accumulator.add_proof(msg.clone()) {
+        if let Some(mut signed_msg) = self
+            .sig_accumulator
+            .add_proof(msg.clone(), &self.public_key_set())
```

Comment:
But the `PublicKeySet` is not intended to be sent with the message. The message only contains the `PublicKey` - but the key itself doesn't carry any information about the threshold or about how to combine the signatures, so the `PublicKeySet` is needed to check whether there are enough signature shares to obtain the full signature.

>however, when all the signatures are received, the receiver may have changed public key set
This is true and it might be the source of some problems I'm seeing in tests. I don't think we should put the `PublicKeySet` in the message, though... unless, maybe, just in the `Partial` variant, as it's the only moment when it's needed?

Reply:
Yes, if since need the the public key set, then the Partial variant seems a reasonable place then.